testnet/src/neon_node.rs

```diff
@@ -322,9 +334,19 @@ fn spawn_miner_relayer(mut relayer: Relayer, local_peer: LocalPeer,
                     }
                 },
                 RelayerDirective::RunTenure(registered_key, last_burn_block) => {
-                    last_mined_block = InitializedNeonNode::relayer_run_tenure(
-                        registered_key, &mut chainstate, &burndb, last_burn_block,
-                        &mut keychain, &mut mem_pool, burn_fee_cap, &mut bitcoin_controller)
+                    last_mined_block =
+                        if genesis_executed {
+                            InitializedNeonNode::relayer_run_tenure(
+                                registered_key, &mut chainstate, &burndb, last_burn_block,
+                                &mut keychain, &mut mem_pool, burn_fee_cap, &mut bitcoin_controller)
+                        } else {
+                            InitializedNeonNode::relayer_run_genesis_tenure(
```

Comment:
There's no way this could get called twice, right?

Reply:
It will get called a second time if the first one didn't win a sortition.

Reply:
But that should be okay -- it would just be a reorg of the fist mined block, not the unmined genesis.